c7n/resolver.py

```diff
@@ -40,6 +41,7 @@ def resolve(self, uri):
 
         if self.cache:
             self.cache.save(("uri-resolver", uri), contents)
+            self.cache.close()
```

Comment:
May I know why we close the cache here? With this, one of my cases runs into an error.

Looking into the below log, the steps are
1. try to get `value-from` value from the cache but not found (at this point, cache loaded)
2. during loading `value-from` value from the file, `get` and `save` the `uri-resolver` value (at this point, cache closed)
3. finally when come to save `value-from` value, raise error

I wonder why not call `self.load()` in cache.py everytime while entering `get()` and `save()`?

```
❯ custodian run -s log/aws --var aws/accounts/nca-vars.yml -c aws/policies/monitor-service-quota.yml -p monitor-service-quota-usage -r us-east-1
init::conn <sqlite3.Connection object at 0x10cc9b300>
get::conn <sqlite3.Connection object at 0x10cc9b300> key ('value-from', ['file:log/aws/monitor-service-quota-usage-prewarm/resources.json', None, '[]|length(@)'])
get::conn <sqlite3.Connection object at 0x10cc9b300> key ('uri-resolver', 'file:log/aws/monitor-service-quota-usage-prewarm/resources.json')
save::key ('uri-resolver', 'file:log/aws/monitor-service-quota-usage-prewarm/resources.json')
save::conn <sqlite3.Connection object at 0x10cc9b300>
resolve::close cache after saved ('uri-resolver', 'file:log/aws/monitor-service-quota-usage-prewarm/resources.json')
save::key ('value-from', ['file:log/aws/monitor-service-quota-usage-prewarm/resources.json', None, '[]|length(@)'])
save::conn None
2022-09-02 10:06:27,526: custodian.commands:ERROR Error while executing policy monitor-service-quota-usage, continuing
Traceback (most recent call last):
  File "/Users/ouj/repos/ncttools/nct-cloud-custodian/lib/cloud-custodian/c7n/commands.py", line 302, in run
    policy()
  File "/Users/ouj/repos/ncttools/nct-cloud-custodian/lib/cloud-custodian/c7n/policy.py", line 1298, in __call__
    elif not self.is_runnable():
  File "/Users/ouj/repos/ncttools/nct-cloud-custodian/lib/cloud-custodian/c7n/policy.py", line 1132, in is_runnable
    return self.conditions.evaluate(event)
  File "/Users/ouj/repos/ncttools/nct-cloud-custodian/lib/cloud-custodian/c7n/policy.py", line 1052, in evaluate
    state = all([f.process([policy_vars], event) for f in self.filters])
  File "/Users/ouj/repos/ncttools/nct-cloud-custodian/lib/cloud-custodian/c7n/policy.py", line 1052, in <listcomp>
    state = all([f.process([policy_vars], event) for f in self.filters])
  File "/Users/ouj/repos/ncttools/nct-cloud-custodian/lib/cloud-custodian/c7n/filters/core.py", line 559, in process
    return super(ValueFilter, self).process(resources, event)
  File "/Users/ouj/repos/ncttools/nct-cloud-custodian/lib/cloud-custodian/c7n/filters/core.py", line 191, in process
    return list(filter(self, resources))
  File "/Users/ouj/repos/ncttools/nct-cloud-custodian/lib/cloud-custodian/c7n/filters/core.py", line 546, in __call__
    matched = self.match(i)
  File "/Users/ouj/repos/ncttools/nct-cloud-custodian/lib/cloud-custodian/c7n/filters/core.py", line 579, in match
    self.v = values.get_values()
  File "/Users/ouj/repos/ncttools/nct-cloud-custodian/lib/cloud-custodian/c7n/resolver.py", line 175, in get_values
    self.cache.save(("value-from", key), contents)
  File "/Users/ouj/repos/ncttools/nct-cloud-custodian/lib/cloud-custodian/c7n/cache.py", line 163, in save
    with self.conn as cursor:
AttributeError: __enter__
2022-09-02 10:06:27,527: custodian.commands:ERROR The following policies had errors while executing
 - monitor-service-quota-usage
```